test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/NotExpressionTests.cs

```diff
@@ -0,0 +1,112 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Microsoft.Health.Fhir.Client;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Microsoft.Health.Fhir.Tests.E2E.Common;
+using Microsoft.Health.Test.Utilities;
+using Xunit;
+using ResourceType = Hl7.Fhir.Model.ResourceType;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest.Search
+{
+    [Trait(Traits.Category, Categories.Search)]
+    [HttpIntegrationFixtureArgumentSets(DataStore.SqlServer, Format.Json)]
+    public sealed class NotExpressionTests : IClassFixture<HttpIntegrationTestFixture>
+    {
+        private readonly TestFhirClient _client;
+
+        public NotExpressionTests(HttpIntegrationTestFixture fixture)
+        {
+            _client = fixture.TestFhirClient;
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        public async Task GivenANotExpressionPattern_WhenSearched_ThenCompareResponseWithExpectedResults()
+        {
+            IReadOnlyList<HealthRecordIdentifier> healthRecordIdentifiers = await GetHealthRecordIdentifiersAsync(CancellationToken.None);
+
+            var practitionerRoles = healthRecordIdentifiers.Where(i => i.ResourceType == ResourceType.PractitionerRole.ToString());
+            foreach (HealthRecordIdentifier practitioner in practitionerRoles)
+            {
+                string query1 = $"_id={practitioner.Id}&active:not=false";
+                Bundle queryResult1 = await _client.SearchAsync(ResourceType.PractitionerRole, query1);
+                Assert.Single(queryResult1.Entry);
+
+                string query2 = $"_id={practitioner.Id}&active:not=false&location:missing=false";
+                Bundle queryResult2 = await _client.SearchAsync(ResourceType.PractitionerRole, query2);
+                Assert.Single(queryResult2.Entry);
+
+                string query3 = $"_id={practitioner.Id}&active:not=true";
+                Bundle queryResult3 = await _client.SearchAsync(ResourceType.PractitionerRole, query3);
+                Assert.Empty(queryResult3.Entry);
+            }
+
+            var healthCareServices = healthRecordIdentifiers.Where(i => i.ResourceType == ResourceType.HealthcareService.ToString());
+            foreach (HealthRecordIdentifier healthCareService in healthCareServices)
+            {
+                // Investigating the possibility of having a bug with :not and chain queries.
```

Comment:
Additional investigation that I'm running: is it possible to have a chain query expression with the :not operator?